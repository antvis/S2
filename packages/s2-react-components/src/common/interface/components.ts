export interface BaseComponentProps<T> {
  /**
   * 标题
   * @default "文字对齐"
   */
  title?: React.ReactNode;

  /**
   * 默认是否折叠
   * @default false
   */
  defaultCollapsed?: boolean;

  /**
   * 默认配置
   */
  defaultOptions?: Partial<T>;

  /**
   * 子节点
   */
  children?: React.ReactNode;
}
