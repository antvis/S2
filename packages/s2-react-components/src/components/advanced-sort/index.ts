export { AdvancedSort } from './advanced-sort';
export { CustomSort } from './custom-sort';

export type {
  AdvancedSortBaseProps,
  AdvancedSortProps,
  CustomSortProps,
  Dimension,
  RuleItem,
  RuleOption,
  RuleValue,
} from './interface';
