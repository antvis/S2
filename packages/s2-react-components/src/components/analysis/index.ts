export * from './advanced-sort';
export * from './drill-down';
export * from './export';
export * from './switcher';
