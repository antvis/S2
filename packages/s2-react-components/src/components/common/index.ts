export { ResetButton } from './reset-button';
export { ResetGroup } from './reset-group';
