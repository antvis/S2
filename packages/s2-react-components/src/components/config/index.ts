export {
  FrozenInputNumber,
  FrozenPanel,
  type FrozenPanelOptions,
  type FrozenPanelProps,
} from './frozen-panel';

export {
  TextAlignPanel,
  type TextAlignPanelOptions,
  type TextAlignPanelProps,
} from './text-align-panel';

export {
  ColorBox,
  ColorPickerPanel,
  ThemePanel,
  type ThemePanelOptions,
  type ThemePanelProps,
} from './theme-panel';
