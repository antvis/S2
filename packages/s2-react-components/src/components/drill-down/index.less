@import '@antv/s2/esm/shared/styles/drill-down.less';
