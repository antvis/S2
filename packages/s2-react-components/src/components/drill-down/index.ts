export { DrillDown } from './drill-down';

export type { DrillDownDataSet, DrillDownProps } from './interface';
