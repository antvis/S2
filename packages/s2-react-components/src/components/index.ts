export * from './common';

export * from './advanced-sort';
export * from './drill-down';
export * from './export';
export * from './switcher';

export * from './frozen-panel';
export * from './text-align-panel';
export * from './theme-panel';
