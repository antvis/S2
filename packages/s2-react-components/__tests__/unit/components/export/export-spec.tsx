import { Export, StrategyExport } from '@/components';
import type { BaseFacet } from '@antv/s2';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { getMockSheetInstance } from '../../../../../s2-react/__tests__/util/helpers';

describe('Export Component Tests', () => {
  const sheetInstance = getMockSheetInstance();

  sheetInstance.options = {
    interaction: {
      copy: {
        enable: true,
      },
    },
  };

  sheetInstance.facet = {
    getRowLeafNodes: jest.fn(() => []),
    getColLeafNodes: jest.fn(() => []),
    getLayoutResult: jest.fn(() => ({})),
  } as unknown as BaseFacet;

  sheetInstance.dataCfg = {
    data: [],
    fields: {
      rows: [],
      columns: [],
      values: [],
    },
  };

  test('should render default component', () => {
    const { asFragment, container } = render(
      <Export className="test" sheetInstance={sheetInstance} />,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('.anticon-more')).toBeDefined();
    expect(container.querySelector('.test')).toBeDefined();
  });

  test('should render component for strategy export', () => {
    const { asFragment, container } = render(
      <StrategyExport sheetInstance={sheetInstance} />,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('.anticon-more')).toBeDefined();
  });

  test('should render custom children', () => {
    const { asFragment } = render(
      <Export sheetInstance={sheetInstance} dropdown={{ open: true }}>
        <div>入口</div>
      </Export>,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('入口')).toBeDefined();
  });

  test('should trigger copy handler', async () => {
    const onCopySuccess = jest.fn();

    render(
      <Export
        sheetInstance={sheetInstance}
        onCopySuccess={onCopySuccess}
        dropdown={{ open: true }}
      />,
    );

    fireEvent.click(screen.getByText('复制原始数据'));
    fireEvent.click(screen.getByText('复制格式化数据'));

    await waitFor(() => {
      expect(onCopySuccess).toHaveBeenCalledTimes(2);
    });
  });

  test('should trigger export handler', async () => {
    const onDownloadSuccess = jest.fn();

    render(
      <Export
        sheetInstance={sheetInstance}
        onDownloadSuccess={onDownloadSuccess}
        dropdown={{ open: true }}
      />,
    );

    fireEvent.click(screen.getByText('下载原始数据'));
    fireEvent.click(screen.getByText('下载格式化数据'));

    await waitFor(() => {
      expect(onDownloadSuccess).toHaveBeenCalledTimes(2);
    });
  });
});
