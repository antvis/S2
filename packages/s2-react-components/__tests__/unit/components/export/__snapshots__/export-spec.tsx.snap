// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Component Tests should render component for strategy export 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-text ant-dropdown-trigger antv-s2-export"
    type="button"
  >
    <span
      aria-label="more"
      class="anticon anticon-more"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`Export Component Tests should render custom children 1`] = `
<DocumentFragment>
  <div
    class="ant-dropdown-trigger ant-dropdown-open antv-s2-export"
  >
    入口
  </div>
</DocumentFragment>
`;

exports[`Export Component Tests should render default component 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-text ant-dropdown-trigger antv-s2-export test"
    type="button"
  >
    <span
      aria-label="more"
      class="anticon anticon-more"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;
